packages/blade/src/components/TopNav/TabNav/utils.ts

```diff
@@ -1,51 +1,40 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
 /* eslint-disable consistent-return */
 import React from 'react';
 import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
 
 /**
- * Check if an element has scroll overflow
+ * Hook to observe resize events on a given element
  */
-const useHasOverflow = (
+const useResize = (
   ref: React.RefObject<HTMLDivElement>,
-  callback?: (hasOverflow: boolean) => void,
-): boolean => {
+  callback?: (entry: ResizeObserverEntry, observer: React.RefObject<ResizeObserver>) => void,
+) => {
   const observer = React.useRef<ResizeObserver | null>(null);
-  const [hasOverflow, setHasOverflow] = React.useState<boolean>(false);
 
   useIsomorphicLayoutEffect(() => {
     if (!ref.current) return;
     const element = ref.current;
 
-    const trigger = (): void => {
-      const hasOverflow = element.scrollWidth > element.clientWidth;
-      setHasOverflow(hasOverflow);
+    if (!('ResizeObserver' in window)) return;
 
-      if (callback) callback(hasOverflow);
-    };
-
-    trigger();
-    if ('ResizeObserver' in window) {
-      observer.current = new ResizeObserver(trigger);
-      observer.current.observe(element);
-    }
+    observer.current = new ResizeObserver((entries) => {
```

Comment:
> isn't this creating new resize observer on every callback change

That should be fine right? That's what we want, otherwise the `callback` function will go stale.

Reply:
Tho we can actually remove the ref and inline this within the effect.